homeassistant/components/hive/config_flow.py

```diff
@@ -90,24 +92,48 @@ async def async_step_2fa(self, user_input=None):
 
             if not errors:
                 try:
+                    if self.context["source"] == config_entries.SOURCE_REAUTH:
+                        return await self.async_setup_hive_entry()
                     self.device_registration = True
-                    return await self.async_setup_hive_entry()
+                    return await self.async_step_configuration()
```

Comment:
This line can not raise UnknownHiveError. Please only wrap the line that can raise in the try... except block.